convert2rhel/pkghandler.py

```diff
@@ -1144,52 +1111,147 @@ def clean_yum_metadata():
-    # complete yet.  will be done in a future work. Right now, all the list of
-    # changes are listed in this comment:
-    # https://github.com/oamg/convert2rhel/pull/469#discussion_r873971400
-    pkg_ver_components = []
-    for pkg in pkg_1, pkg_2:
-        # Remove the package name and split the rest between epoch + version
-        # and release + arch
-        epoch_version, release_arch = pkg.rsplit("-", 2)[-2:]
-        # Separate the (optional) epoch from the version
-        epoch_version = epoch_version.split(":", 1)
-        if len(epoch_version) == 1:
-            epoch = "0"
-            version = epoch_version[0]
+def _parse_pkg_with_yum(pkg):
+    """
+    Parse verison string using yum and rpmUtils splitFilename
+    :return: Return a list containing name, epoch, version, release, arch
+    """
+
+    # package is in NEVRA/NEVR format
+    if re.findall(r"-\d:", pkg):
+        name, epoch_version, release_arch = pkg.rsplit("-", 2)
+        epoch, version = epoch_version.split(":", 1)
+        # if NEVRA format
+        if re.findall(r".*(x86_64|s390x|i86|ppc64le|aarch64)$", release_arch):
+            release, arch = release_arch.rsplit(".", 1)
+        # if NEVR format
         else:
-            epoch, version = epoch_version
-        # Discard the arch
-        release = release_arch.rsplit(".", 1)[0]
-        pkg_ver_components.append((epoch, version, release))
+            release = release_arch
+            arch = None
+
+    # package is in ENVR, ENVRA, NVR, NVRA
+    else:
+        (name, version, release, epoch, arch) = pkgmanager.splitFilename(pkg)
+
+        # splitFilename places part of the release in the arch field when no arch is present
+        # if arch field is invalid, append contents to release field
+        if arch not in ("x86_64", "s390x", "i86", "ppc64le", "aarch64"):
+            temp_release = arch
+            arch = None
+            release = release + "." + temp_release
+
+    pkg_ver_components = [name, epoch, version, release, arch]
+    # convert any empty strings to None for consistency
+    pkg_ver_components = [str(i or None) for i in pkg_ver_components]
+
+    return pkg_ver_components
+
+
+def _parse_pkg_with_dnf(pkg):
+    """
+    Parse verison string using hawkey and dnf
+    :return: Return a list containing name, epoch, version, release, arch
+    """
+
+    name = epoch = version = release = arch = None
+    no_arch_data = None
+
+    # if format is ENVR/ENVRA, store and remove epoch and evaluate as NVR/NVRA
+    if re.findall(r"^\d:", pkg):
+        pkg = pkg.split(":")
+        epoch = pkg[0]
+        pkg = pkg[1]
+
+    # returns generator for every possible nevra
+    subject = pkgmanager.dnf.subject.Subject(pkg)
+    possible_nevra = subject.get_nevra_possibilities(forms=[pkgmanager.hawkey.FORM_NEVRA, pkgmanager.hawkey.FORM_NEVR])
+
+    # loop through each possible set of nevra fields and select the valid one
+    for i, nevra in enumerate(possible_nevra):
```

Comment:
The `i` is not being used 
```suggestion
    for _, nevra in enumerate(possible_nevra):
```